cirq/contrib/quimb/density_matrix.py

```diff
@@ -0,0 +1,195 @@
+from functools import lru_cache
+from typing import Sequence, Dict, Union, Tuple, List
+
+import numpy as np
+import quimb
+import quimb.tensor as qtn
+
+import cirq
+
+
+@lru_cache()
+def _qpos_tag(qubits: Union[cirq.LineQubit, Tuple[cirq.LineQubit]]):
+    """Given a qubit or qubits, return a "position tag" (used for drawing).
+
+    For multiple qubits, the tag is for the first qubit.
+    """
+    if isinstance(qubits, cirq.LineQubit):
+        return _qpos_tag((qubits,))
+    x = min(q.x for q in qubits)
+    return f'q{x}'
+
+
+@lru_cache()
+def _qpos_y(qubits: Union[cirq.LineQubit, Tuple[cirq.LineQubit]],
+            n_qubits: int):
```

Comment:
isn't n_qubits == len(qubits)?

Reply:
No. Changing `n_qubits` to `tot_n_qubits`, the total number of qubits (needed to turn qubit index into a y position (which are backwards)). `qubits` is usually one or two qubits, corresponding to a tensor, corresponding to a gate operation

Reply:
and adding a docstring